File: MotivApp/app/src/main/java/com/motiv/piotr/dao/PostDao.java
package com.motiv.piotr.dao;

import androidx.fragment.app.*;
import androidx.room.*;
import com.motiv.piotr.PostWithReferences;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;

@Dao
public interface PostDao {

    @Insert
    public long savePost(com.motiv.piotr.Post post);

    @Delete
    public int deletePost(com.motiv.piotr.Post post);

    @Query("SELECT * FROM post")
    @Transaction
    public java.util.List<PostWithReferences> getPosts();

    @Query("SELECT * FROM post WHERE id = :id")
    @Transaction
    public PostWithReferences getPost(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void savePosts(java.util.List<com.motiv.piotr.Post> post);
}

File: MotivApp/app/src/main/java/com/motiv/piotr/dao/PostDao.kt
 
package com.motiv.piotr.dao
import androidx.room.*
import com.motiv.piotr.Post
import com.motiv.piotr.PostWithReferences
import kotlin.collections.List
@Dao
public interface PostDao {

    @Insert
    public fun savePost(post: com.motiv.piotr.Post): Long@Delete
    public fun deletePost(post: com.motiv.piotr.Post): Int@Query("SELECT * FROM post")
    @Transaction
    public fun getPosts(): List<PostWithReferences>@Query("SELECT * FROM post WHERE id = :id")
    @Transaction
    public fun getPost(id: String): PostWithReferences@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun savePosts(post: List<com.motiv.piotr.Post>): Unit
}
